calabash/Classes/FranklyServer/Routes/LPReflectionRoute.m

```diff
@@ -0,0 +1,67 @@
+#if ! __has_feature(objc_arc)
+#warning This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#import "LPReflectionRoute.h"
+#import <objc/runtime.h>
+
+@interface LPReflectionRoute ()
+
+- (NSArray *) libraryNames;
+- (NSArray *) classNames;
+
+@end
+
+@implementation LPReflectionRoute
+
+- (BOOL) supportsMethod:(NSString *) method atPath:(NSString *) path {
+  return [method isEqualToString:@"GET"];
+}
+
+- (NSDictionary *) JSONResponseForMethod:(NSString *) method
+                                     URI:(NSString *) path
+                                    data:(NSDictionary *) data {
+
+  return @{
+           @"libraries" : [self libraryNames],
+           @"classes" : [self classNames]
+           };
+}
+
+- (NSArray *) libraryNames {
+  unsigned int number = 0;
+
+  const char **names = objc_copyImageNames(&number);
+
+  NSMutableArray *array = [NSMutableArray arrayWithCapacity:number];
```

Comment:
It might be possible to modularize this function into a generic `- (NSArray *)names( rawGeneratorBlock )` so that you could use the same function to get both library names and class names, but it's probably more readable this way.